src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/repositories/StudyRepository.java

```diff
@@ -37,4 +39,7 @@
             @Param("patch") String patch
     );
 
+    @RestResource(path = "findByType")
```

Comment:
In order projects we have discussed using `/search` as root endpoint. So this class could have a hierarchy like:

* GET /studies/search/assembly?name=GRCh37
* GET /studies/search/assembly?name=GRCh37&patch=2
* GET /studies/search/type?type=WGS

I would prefer if it was completely flexible as in the following:

* GET /studies/search?assembly-name=GRCh38&assembly-patch=2&type=WGS

This seems doable using query-by-example (see [this](https://github.com/spring-projects/spring-data-examples/tree/master/jpa/query-by-example) and [this](https://stackoverflow.com/questions/44052493/spring-data-jpa-findby-optional-variables)).

What do you think @Zhicheng-Liu, @jorizci and @selvaebi - shall we do it or not? If we do, shall we make it part of this PR or another one to unify all the searches?

Reply:
@cyenyxe @Zhicheng-Liu I'd suggest investigating this other thing https://spring.io/blog/2015/09/04/what-s-new-in-spring-data-release-gosling#querydsl-support

Don't know how well goes with springfox, but seems easy enough to enable it so it could be feasible to include it in this ticket checking it out and deciding.

Reply:
I have now added QueryDsl support and also added RestControllers for studies and assemblies. Now you can filtering, for example, assemblies with URL like /assemblies?name=grch37 or /search/assemblies?name=grch37

Reply:
Changed to use /assemblies?name=grch37 or /assemblies/search?name=grch37 formats.